# Analizando os dados do Dataset Íris 

In [19]:
import pandas as pd
from sklearn import datasets
from sklearn.model_selection import train_test_split

# Aqui dizemos o classificador que iremos usar posteriormente
classifier = 'DecisionTree' # KNN | DecisionTree

# Felizmente, os datasets do módulo de sklearn já vem separados e tratados, então podemos nos preocupar mais no modelo.
iris_base = datasets.load_iris()

In [17]:
X = iris_base.data
y = iris_base.target
len(X)

150

Aqui agora separamos as variáveis de teste das de treinamento:

In [18]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size= 0.33)
len(X_train)

100

Agora vamos treinar o modelo!

In [20]:
# Primeiro precisamos importar um classificador, ou seja, o que criará nosso modelo.
if classifier == 'DecisionTree':
    from sklearn import tree
    model = tree.DecisionTreeClassifier()
else:
    from sklearn import neighbors
    model = neighbors.KNeighborsClassifier()

In [21]:
# Aqui treinamos nosso modelo
model.fit(X_train, y_train)

DecisionTreeClassifier()

In [22]:
# Agora pegamos os valores que usamos de teste e vamos prever!
pred = model.predict(X_test)

In [25]:
# Agora vamos calcular o MAE (Mean Abslute Error)
from sklearn.metrics import accuracy_score
accuracy_score(y_test, pred)

0.94

Podemos ver aqui que o modelo tem uma acurácia de 94%, o que é grande para nossos testes e mundo real.